lib/ui/layout.js

```diff
@@ -1,101 +1,87 @@
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.debuggerPane')) panes.push(debuggerPane)
+  if (atom.config.get('julia-client.uiOptions.layouts.defaultPanes.linter')) panes.push(linter)
+
+  openPanes(panes)
+
+  // Re-activate the initially activated item
+  activeContainer.activate()
+  activePane.activate()
+  activePane.activateItem(activeItem)
 }
 
-function workspace() { return require('../runtime').workspace.ws }
-function docs() { return require('../ui').docpane.pane }
-function cons() {
-  if (atom.config.get('julia-client.consoleOptions.consoleStyle') == 'REPL-based') {
-    return require('../runtime').console2.terminal
-  } else {
-    return require('../runtime').console.c
+function openPanes (panes) {
+  if (panes.length > 0) {
+    const pane = panes.shift()
+    const promise = pane().open()
+    promise.then(() => {
+      // @NOTE: Focus the previously focused pane after each opening in order for this command to
+       // behave as if an user dispatch each opening command/action separately.
+       // In other word, this prevents opening multiple panes with the same splitting rule in a
+       // same location from ending up in a funny state
+      const container = atom.workspace.getActivePaneContainer()
+      container.activatePreviousPane()
+      openPanes(panes)
+    })
   }
+  return
 }
-function plots() { return require('../runtime').plots.pane }
 
-
-export function standard() {
-  let layout = {
-    split: true,
-    vertical: false,
-    items: [
-      null,
+export function queryDefaultLayout () {
+  const message = atom.notifications.addInfo('Julia-Client: Open Juno-specific panes on startup ?', {
+    buttons: [
+      {
+        text: 'Yes',
+        onDidClick: () => {
+          defaultLayout()
+          message.dismiss()
+          atom.config.set('julia-client.firstBoot', false)
+          atom.config.set('julia-client.uiOptions.layouts.openDefaultPanesOnStartUp', true)
+        }
+      },
       {
-        split: false,
-        items: [workspace(), docs(), plots()]
+        text: 'No',
+        onDidClick: () => {
+          message.dismiss()
+          atom.config.set('julia-client.firstBoot', false)
+          atom.config.set('julia-client.uiOptions.layouts.openDefaultPanesOnStartUp', false)
+        }
       }
-    ]
-  }
-
-  resetLayout()
-
-  let centralpane = atom.workspace.getActivePane()
-
-  let ps = []
-  for (let pane of [workspace(), cons(), docs(), plots()]) ps.push(pane.close())
-  Promise.all(ps).then(() => {
-    resetLayout()
-    createLayout(layout)
-    atom.commands.dispatch(atom.views.getView(atom.workspace.getActivePane()), 'julia-client:open-console')
-    if (centralpane.activate) centralpane.activate()
+    ],
+    description:
+      `You can specify the panes to be opened and their _default location_ and _splitting rule_ in
+       **'Settings -> Julia-Client > UI Options > Layout'**.\n
```

Comment:
Maybe  put the settings path in backticks?

Reply:
I tried that with `[**'Settings -> Julia-Client > UI Options > Layout'**](atom://config/packages/julia-client)]` but that didn't work 😢 
Any other idea ?

Reply:
You could add a button that opens the settings, but I don't think that's worth it. Imho 
```
`Settings -> Julia-Client > UI Options > Layout`.\n
```
would just look better, but it's not super important.

Reply:
Yeah, I also wondered about adding the button, but I thought that would make the notification looks more complicated, thus I didn't do that.

If you're okay, I want to let this go for now.

Reply:
Yeah, let's keep it like this for now.

Reply:
Ah, I totally misunderstood what you meant ! Backticks !
Yeah, it should look better, thus I changed it in the recent commit 😉